frontend/src/components/common/InviteButton.vue

```diff
@@ -0,0 +1,228 @@
+    </h3>
+
+    <ul class="mb-4">
+      <li>If a user already has permissions, you can link them with a share link</li>
+      <li>Invite someone using an invite link</li>
+      <li>To share publicly or with a direct link, you must set the file to public - this only works for objects</li>
+    </ul>
+    <TabView>
+      <TabPanel
+        v-if="obj?.public"
+        header="Direct public file link"
+      >
+        <ShareLinkContent
+          :share-link="comsUrl"
+          label="Direct Link"
+        />
+      </TabPanel>
+      <!-- Disable for public until unauthed File Details page works -->
+      <TabPanel
+        header="Share link"
+        :disabled="obj?.public"
+      >
+        <ShareLinkContent
+          :share-link="bcBoxLink"
+          label="Share Link"
+        />
+      </TabPanel>
+      <TabPanel header="Invite link">
+        <h3 class="mt-1 mb-2">{{ props.labelText }} Invite</h3>
+        <p>Make invite available for:</p>
+        <div class="flex flex-wrap gap-3">
+          <div
+            v-for="t in timeFrames"
+            :key="t.value.toString()"
+            class="flex align-items-center"
+          >
+            <RadioButton
+              v-model="formData.expiresAt"
+              :input-id="t.value.toString()"
+              :name="t.name"
+              :value="t.value"
+            />
+            <label
+              :for="t.value.toString()"
+              class="ml-2"
+            >
+              {{ t.name }}
+            </label>
+          </div>
+        </div>
+        <p class="mt-4 mb-2">Restrict to user's email</p>
+        <div class="p-inputgroup mb-4">
+          <InputSwitch
+            v-model="isRestricted"
+            class="mt-1"
+          />
+          <!-- Add email validation -->
+          <InputText
+            v-show="isRestricted"
+            v-model="formData.email"
+            name="inviteEmail"
+            placeholder="Enter email"
+            required
+            type="email"
+            class="ml-5 mr-8"
+          />
+        </div>
+        <div class="p-inputgroup mb-4">
+          <Button
+            class="p-button p-button-primary"
+            @click="sendInvite"
+          >
+            <font-awesome-icon
+              icon="fa fa-envelope"
+              class="mr-2"
+            />
+            Generate invite link
+          </Button>
+        </div>
+
+        <ShareLinkContent
+          v-if="showInviteLink"
+          :share-link="inviteLink"
+          label="Invite Link"
+        />
+      </TabPanel>
+    </TabView>
+  </Dialog>
+  <Button
+    v-tooltip.bottom="`${props.labelText} Share`"
+    class="p-button-lg p-button-text primary"
+    aria-label="`${props.labelText} Share`"
+    @click="displayInviteDialog = true"
+  >
+    <font-awesome-icon icon="fa-solid fa-share-alt" />
+  </Button>
+</template>
+
+<style scoped lang="scss">
+h2 {
+  font-weight: bold;
+}
+ul {
+  padding-left: 22px;
+}
```

Comment:
While these style changes suffice to get the job done, it would be better to instead assign a class or self-made construct to modify these styles instead of modifying the base type itself - in the future if this component were to be expanded and reuse these same types, they could cause some undesired effects for styling.